homeassistant/components/homeconnect/switch.py

```diff
@@ -0,0 +1,189 @@
+        try:
+            self.device.appliance.set_setting(
+                "BSH.Common.Setting.PowerState", "BSH.Common.EnumType.PowerState.On"
+            )
+        except HomeConnectError as err:
+            _LOGGER.error("Error while trying to turn on device: %s", err)
+            self._state = False
+        self.async_entity_update()
+
+    def turn_off(self, **kwargs):
+        """Switch the device off."""
+        _LOGGER.debug("tried to switch off %s", self.name)
+        try:
+            self.device.appliance.set_setting(
+                "BSH.Common.Setting.PowerState", self.device.power_off_state
+            )
+        except HomeConnectError as err:
+            _LOGGER.error("Error while trying to turn off device: %s", err)
+            self._state = True
+        self.async_entity_update()
+
+    def update(self):
+        """Update the switch's status."""
+        if (
+            self.device.appliance.status.get("BSH.Common.Setting.PowerState", {}).get(
+                "value", None
+            )
+            == "BSH.Common.EnumType.PowerState.On"
+        ):
+            self._state = True
+        elif (
+            self.device.appliance.status.get("BSH.Common.Setting.PowerState", {}).get(
+                "value", None
+            )
+            == self.device.power_off_state
+        ):
+            self._state = False
+        elif self.device.appliance.status.get(
+            "BSH.Common.Status.OperationState", {}
+        ).get("value", None) in [
+            "BSH.Common.EnumType.OperationState.Ready",
+            "BSH.Common.EnumType.OperationState.DelayedStart",
+            "BSH.Common.EnumType.OperationState.Run",
+            "BSH.Common.EnumType.OperationState.Pause",
+            "BSH.Common.EnumType.OperationState.ActionRequired",
+            "BSH.Common.EnumType.OperationState.Aborting",
+            "BSH.Common.EnumType.OperationState.Finished",
+        ]:
+            self._state = True
+        elif (
+            self.device.appliance.status.get(
+                "BSH.Common.Status.OperationState", {}
+            ).get("value", None)
+            == "BSH.Common.EnumType.OperationState.Inactive"
+        ):
+            self._state = False
+        else:
+            self._state = None
+        _LOGGER.debug("Updated, new state: %s", self._state)
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        status = self.device.appliance.status
+        attributes = {}
+        for key, val in status.items():
+            # e.g. `BSH.Something.SomeValue` -> `some_value`
+            newkey = convert_to_snake(key.split(".")[-1])
+            if isinstance(val, str):
+                newval = convert_to_snake(val.split(".")[-1])
+            elif isinstance(val, (dict, int)):
+                newval = val
+            else:
+                raise ValueError("Unexpected type for value")
+            attributes[newkey] = newval
```

Comment:
We prefer to set attribute keys explicitly so we know what they are and can check formatting.

Can we create a dictionary that maps the attribute keys to the status keys, and look up that dictionary using the status key before setting the state attributes here?

Reply:
The problem is that I don't really know which attributes are reported by different types of devices and it was frequently requested by users of the custom integration to get access to some attribute (oven temperature, hood light status, stuff like that) for which there is no specific sensor, so it seemed to me to make sense to add all of them so users can create template sensors for whatever they need.

Reply:
Normally we want measurements exposed as individual sensor entities. That's also a reason why we want to review all state attributes. If we don't know what the measurements are we can not add them.

Reply:
Yes, that makes sense. Adding more sensors is something for the future, but I currently cannot realistically research and test all the relevant measurements. So should I remove the attributes altogether then for the time being?

Reply:
Yeah, that's best for now. :+1: